api/src/controllers/donate-device.controller.ts

```diff
@@ -56,6 +60,66 @@ export class DonateDeviceController {
     return this.donateDeviceRepository.create(donateDevice);
   }
 
+  @post('/donate-devices-corporate')
+  @response(200, {
+    description: 'Corporate model instance',
+    content: {'application/json': {schema: getModelSchemaRef(Corporate)}},
+  })
+  async createCorporate(
+    @requestBody({
+      content: {
+        'application/json': {
+          schema: getModelSchemaRef(Corporate, {
+            title: 'NewSurvay',
+            exclude: ['id'],
+          }),
+        },
+      },
+    })
+    corporateResponce: Omit<Corporate, 'id'>,
+  ): Promise<Corporate> {
+    let instanceID = corporateResponce.data[0]?.instanceID;
+    instanceID = instanceID
+      .split(':')?.[1]
+      ?.split('-')?.[0]
+      .toUpperCase();
+    const data = corporateResponce?.data?.[0];
+    data.instanceID = instanceID;
+    const corporateType = new CorporateType(data);
+    const gQLHelper = new graphQLHelper();
+
+    if (corporateType.poc_phone_number) {
+      const {errors, data: gqlResponse} = await gQLHelper.startExecuteInsert(
+        corporateType,
+      );
+      if (errors) {
+        console.error(errors);
+      } else {
+        let trackingKeys: String[] = [];
+        for(let i = 0; i < gqlResponse.insert_device_donation_corporates_one.quantity_of_devices; i++ ) {
+          let r = (Math.random() + 1).toString(36).substring(4).toUpperCase();
+          trackingKeys = [...trackingKeys,r];
+          const corporateDevicesType = new CorporateDevicesType({"trackingKey":r,'companyId':gqlResponse.insert_device_donation_corporates_one.company_id});
+          const {errors, data: cdGqlResponse} = await gQLHelper.startExecuteInsert(
+            corporateDevicesType,
+          );
+          if (errors) {
+            console.error(errors);
+          } else {
+            console.log(cdGqlResponse);
+          }
+        }
+        const smsBody = `Congratulations! You have successfully registered for donating <Number of devices donated> smartphones as part of the ""Digital Saathi"" campaign.
+        \nPlease note your tracking IDs: ${trackingKeys}. You can use these IDs to track the status of delivery for your donated smartphones. Contact 1800-180-8190 for any assistance.
```

Comment:
- The "<Number of devices donated>" should be replaced by `quantitiy_of_devices`
- The tracking ID should be `company_id`